### PR TITLE
net-misc/rygel: version bump to 0.42.0, net-libs/gssdp: vbump to 1.6.0, net-libs/gupnp: vbump to 1.6.0

### PR DESCRIPTION
**Pull request by Schievel1**
gssdp-1.6 has a file conflict with gssdp-1.4, thats why I remove that file from 1.6 and depend on 1.4

### CODE REVIEW

**Comment by mattst88** [comment]
> [net-misc/gnome-remote-desktop: fix 872302](https://github.com/gentoo/gentoo/pull/27472/commits/5c7efe080c55ccc11a6dc23f482ea1543bb0a94f)

Please drop this commit from the PR.

> [net-misc/rygel: bump to version 0.42.0](https://github.com/gentoo/gentoo/pull/27472/commits/a6fc206d038852c44911de9e62c64d3b68aaecf8)

Please split this commit into one commit per package.

File: net-libs/gssdp/gssdp-1.6.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
```
Comment (mattst88) [inline net-libs/gssdp/gssdp-1.6.0.ebuild:4]: EAPI=8

File: net-libs/gssdp/gssdp-1.6.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+VALA_USE_DEPEND="vapigen"
```
Comment (mattst88) [inline net-libs/gssdp/gssdp-1.6.0.ebuild:5]: Drop

File: net-libs/gupnp/gupnp-1.6.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
```
Comment (mattst88) [inline net-libs/gupnp/gupnp-1.6.0.ebuild:4]: EAPI=8

File: net-libs/gupnp/gupnp-1.6.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+VALA_USE_DEPEND="vapigen"
```
Comment (mattst88) [inline net-libs/gupnp/gupnp-1.6.0.ebuild:5]: Drop

File: net-misc/rygel/rygel-0.42.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,83 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+VALA_MIN_API_VERSION="0.53.2"
```
Comment (mattst88) [inline net-misc/rygel/rygel-0.42.0.ebuild:5]: Drop

File: net-misc/rygel/rygel-0.42.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,83 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (mattst88) [inline net-misc/rygel/rygel-0.42.0.ebuild:4]: EAPI=8

File: net-libs/gupnp/gupnp-1.6.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+REQUIRED_USE="${PYTHON_REQUIRED_USE}
+	?? ( connman networkmanager )
+	gtk-doc? ( introspection )
+"
+
+# prefix: uuid dependency can be adapted to non-linux platforms
+RDEPEND="${PYTHON_DEPS}
+	>=dev-libs/glib-2.69:2[${MULTILIB_USEDEP}]
+	>=net-libs/gssdp-1.5.2:1.6=[introspection?,${MULTILIB_USEDEP}] # TODO
+	>=net-libs/libsoup-2.48.0:2.4[introspection?,${MULTILIB_USEDEP}]
```
Comment (mattst88) [inline net-libs/gupnp/gupnp-1.6.0.ebuild:28]:
````diff
-    dependency('libsoup-2.4', version : '>= 2.48.0'),
+    dependency('libsoup-3.0', version : '>= 2.99.0'),
````

File: net-libs/gupnp/gupnp-1.6.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+	>=dev-libs/libxml2-2.9.1-r4[${MULTILIB_USEDEP}]
```
Comment (mattst88) [inline net-libs/gupnp/gupnp-1.6.0.ebuild:29]: This has been moved right after gssdp in meson.build.

File: net-libs/gssdp/gssdp-1.6.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+DESCRIPTION="GObject-based API for handling resource discovery and announcement over SSDP"
+HOMEPAGE="https://wiki.gnome.org/Projects/GUPnP https://gitlab.gnome.org/GNOME/gssdp"
+
+LICENSE="LGPL-2+"
+SLOT="1.6/1.6-0" # <API version>-<soname> # TODO what is the right slot for this?
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~loong ~ppc ~ppc64 ~riscv ~sparc ~x86"
+IUSE="gtk-doc +introspection gtk vala"
+REQUIRED_USE="
+	vala? ( introspection )
+	gtk-doc ( introspection )
```
Comment (mattst88) [inline net-libs/gssdp/gssdp-1.6.0.ebuild:18]: `gtk-doc*?*`

Comment (mattst88) [reply net-libs/gssdp/gssdp-1.6.0.ebuild:18]: Also, put it before `vala?` (alphabetically)

File: net-misc/rygel/rygel-0.42.0.ebuild (hunk 5 above)
Comment (mattst88) [reply net-misc/rygel/rygel-0.42.0.ebuild:5]: >  *   Invalid syntax of VALA_MIN_API_VERSION

File: net-misc/rygel/rygel-0.42.0.ebuild [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+	sqlite? (
+		>=dev-db/sqlite-3.5:3
+		dev-libs/libunistring:=
+	)
+	>=media-libs/gstreamer-1.12:1.0
+	>=media-libs/gst-plugins-base-1.12:1.0
+	media-libs/gstreamer-editing-services:1.0
+	>=media-libs/libmediaart-0.7:2.0[vala]
+	media-plugins/gst-plugins-soup:1.0
+	x11-libs/gdk-pixbuf:2
```
Comment (mattst88) [inline net-misc/rygel/rygel-0.42.0.ebuild:35]: I don't see gdk-pixbuf as a direct dependency.

**Review by mattst88 — CHANGES_REQUESTED** [review]

**Comment by Schievel1** [comment]
@mattst88 pleaze don't waste your time reviewing a draft PR. I haven't done anything with this PR yet, just committed and put it up

File: net-misc/rygel/rygel-0.42.0.ebuild (hunk 10 above)
Comment (Schievel1) [reply net-misc/rygel/rygel-0.42.0.ebuild:35]: https://gitlab.gnome.org/GNOME/rygel/-/blob/master/meson.build#L104 ?

**Comment by Schievel1** [comment]
I will add the signoffs, I was playing around with using pkgdev commit instead of pkgcommit.

Otherwise this is ready for review

**Comment by mattst88** [comment]
This looks good. I'm just poking at getting rygel's `USE=gtk-doc` working. It appears to need `dev-lang/vala[valadoc]` but even still it's not detected.

File: net-misc/rygel/rygel-0.42.0.ebuild [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+	#sed -e 's/return rygel_playbin_renderer_test_main (argv, argc);/return 0;/' \
+	#	-i tests/rygel-playbin-renderer-test.c || die
+
+	#default
+}
+
+src_configure() {
+	local emesonargs=(
+		$(meson_use gtk-doc api-docs)
+		$(meson_use man man_pages)
```
Comment (mattst88) [inline net-misc/rygel/rygel-0.42.0.ebuild:72]: FYI, we should always build man pages.

**Comment by mattst88** [comment]
I see the dep on `libX11` was dropped, but I actually see that it was added in the `pkgdiff -b` output?

````diff
--- /tmp/mgorny-dev-scripts/portage/net-misc/rygel-0.40.4/work/rygel-0.40.4/src/rygel/meson.build       2022-06-03 15:18:20.000000000 -0400
+++ /tmp/mgorny-dev-scripts/portage/net-misc/rygel-0.42.0/work/rygel-0.42.0/src/rygel/meson.build       2022-09-17 06:44:58.000000000 -0400
@@ -1,11 +1,11 @@
 rygel_sources = [
         'rygel-acl.vala',
-        'rygel-dbus-service.vala',
-        'rygel-main.vala'
+        'application.vala',
+        'rygel-dbus-service.vala'
         ]
 executable('rygel',
            rygel_sources,
            c_args : ['-DG_LOG_DOMAIN="Rygel"'],
-           dependencies : rygel_deps + [build_config, rygel_core, rygel_server],
+           dependencies : rygel_deps + [build_config, rygel_core, rygel_server, dependency('x11')],
            install: true
            )
````

**Comment by mattst88** [comment]
I pushed some fixup commits to the rygel branch of my tree.

**Comment by Schievel1** [comment]
> I see the dep on `libX11` was dropped, but I actually see that it was added in the `pkgdiff -b` output?
> 
> ```diff
> --- /tmp/mgorny-dev-scripts/portage/net-misc/rygel-0.40.4/work/rygel-0.40.4/src/rygel/meson.build       2022-06-03 15:18:20.000000000 -0400
> +++ /tmp/mgorny-dev-scripts/portage/net-misc/rygel-0.42.0/work/rygel-0.42.0/src/rygel/meson.build       2022-09-17 06:44:58.000000000 -0400
> @@ -1,11 +1,11 @@
>  rygel_sources = [
>          'rygel-acl.vala',
> -        'rygel-dbus-service.vala',
> -        'rygel-main.vala'
> +        'application.vala',
> +        'rygel-dbus-service.vala'
>          ]
>  executable('rygel',
>             rygel_sources,
>             c_args : ['-DG_LOG_DOMAIN="Rygel"'],
> -           dependencies : rygel_deps + [build_config, rygel_core, rygel_server],
> +           dependencies : rygel_deps + [build_config, rygel_core, rygel_server, dependency('x11')],
>             install: true
>             )
> ```
Oh man, I added it to the ebuild 0.40.2. I shouldn't work on this before the first coffee hits

**Comment by Schievel1** [comment]
> I pushed some fixup commits to the rygel branch of my tree.

I merged them into my commit. Are you sure `app-text/docbook-xml-dtd:4.5` is necessary? I can merge to package without problems without it. 

It doesn't build gtk-docs though, but I doesn't do so with `app-text/docbook-xml-dtd:4.5` either.

**Comment by mattst88** [comment]
> `app-text/docbook-xml-dtd`

Without it, I see this (non-fatal) warning in the build log when generating man pages:

````
[85/351] /usr/bin/xsltproc --nonet --stringparam man.output.quietly 1 --stringparam funcsynopsis.style ansi --stringparam man.authors.section.enabled 1 --stringparam man.copyright.section.enabled 1 -o doc/man/rygel.1 http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl ../rygel-0.42.0/doc/man/rygel.xml
../rygel-0.42.0/doc/man/rygel.xml:61: warning: failed to load external entity "http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd"
]>
  ^
````

**Comment by Schievel1** [comment]
> > `app-text/docbook-xml-dtd`
> 
> Without it, I see this (non-fatal) warning in the build log when generating man pages:
> 
> ```
> [85/351] /usr/bin/xsltproc --nonet --stringparam man.output.quietly 1 --stringparam funcsynopsis.style ansi --stringparam man.authors.section.enabled 1 --stringparam man.copyright.section.enabled 1 -o doc/man/rygel.1 http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl ../rygel-0.42.0/doc/man/rygel.xml
> ../rygel-0.42.0/doc/man/rygel.xml:61: warning: failed to load external entity "http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd"
> ]>
>   ^
> ```

Well yeah, with all this text flying past this is easy to miss

File: net-misc/rygel/rygel-0.40.4.ebuild [hunk 12]
```diff
@@ -36,13 +37,15 @@ DEPEND="
...
-	tracker? ( app-misc/tracker:3= )
+	tracker? ( app-misc/tracker:3 )
 	transcode? (
 		media-libs/gst-plugins-bad:1.0
 		media-plugins/gst-plugins-twolame:1.0
 		media-plugins/gst-plugins-libav:1.0
 	)
 	gtk? ( >=x11-libs/gtk+-3.22:3 )
+
+	x11-libs/libX11
```
Comment (mattst88) [inline net-misc/rygel/rygel-0.40.4.ebuild:48]: Whaaaa?

The x11 dep was added in https://gitlab.gnome.org/GNOME/rygel/-/commit/18cd2cff6f64b7a54156ac97e5c4f50eb8fd8d67 AFAICT, which is not in v0.40.

In general, we shouldn't be modifying stable ebuilds, and we shouldn't be modifying other ebuilds in version bump commits.

File: net-misc/rygel/rygel-0.42.0.ebuild (hunk 10 above)
Comment (mattst88) [reply net-misc/rygel/rygel-0.42.0.ebuild:35]: Right you are!

File: net-misc/rygel/rygel-0.40.4.ebuild (hunk 12 above)
Comment (Schievel1) [reply net-misc/rygel/rygel-0.40.4.ebuild:48]: oh ffs I accidentally added this in, then I removed this, then did the rebase from your branch and added it again this way.  -.-

**Pull request closed without merging**